lib/streamlit/elements/widgets/button_group.py

```diff
@@ -168,6 +171,28 @@ def _build_proto(
 
 
 class ButtonGroupMixin:
+    @overload  # These overloads are not documented in the docstring, at least not at this time, on the theory that most people won't know what it means. And the Literals here are a subclass of int anyway.
```

Comment:
This won't pass the linter checks, I'm pretty sure. Make sure you run `make pyformat`.

Reply:
I was surprised as well, but it ruff tolerates it. It passes my copy of the CI, which I've enabled on my GitHub repo fork. (The true moment of truth still awaits, of course 🍿)

I think this is because comments are a special case that formatting tools know they shouldn't feel free to break up, and the thing before the comment is atomic so it can't be broken up. So ruff check passes it and ruff format doesn't change it.